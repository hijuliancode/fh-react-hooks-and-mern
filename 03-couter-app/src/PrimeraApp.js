import React from 'react';
import PropTypes from 'prop-types'

// FC

const PrimeraApp = ({ saludo, subtitle }) => {
    
    return (
        <>
            {/* <pre>{ JSON.stringify(persona, null, 2) }</pre> */}
            {/* <hr/> */}
            <h1> {saludo} </h1>
            <p>{subtitle}</p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}
PrimeraApp.defaultProps = {
    subtitle: 'Mi primera aplicaci√≥n'
}

export default PrimeraApp;