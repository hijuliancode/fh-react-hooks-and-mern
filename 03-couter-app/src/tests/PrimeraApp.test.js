import React from 'react';
import { shallow } from 'enzyme';
// import { render } from "@testing-library/react"

import PrimeraApp from "../PrimeraApp"

describe('Pruebas en PrimeraApp', () => {
  // test('should mostrar el mensaje Hola, soy Julian', () => {
  //   const saludo = 'Hola, soy Julian'
  //   // const wrapper = render( <PrimeraApp saludo={saludo} /> )
  //   const { getByText } = render( <PrimeraApp saludo={saludo} /> )
  //   expect( getByText(saludo) ).toBeInTheDocument()
  // })
  test('should mostrar <PrimeraApp/> correctamente', () => {
    const saludo = 'Hola, soy Goku'
    const wrapper = shallow( <PrimeraApp saludo={ saludo } /> )

    expect( wrapper ).toMatchSnapshot();
  })
  
})
