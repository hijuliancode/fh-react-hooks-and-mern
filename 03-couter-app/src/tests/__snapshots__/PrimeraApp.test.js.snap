// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp should mostrar <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola, soy Goku
     
  </h1>
  <p>
    Mi primera aplicaci√≥n
  </p>
</Fragment>
`;
